client/executor/src/native_executor.rs

```diff
@@ -219,6 +219,47 @@ where
 		allow_missing_host_functions: bool,
 		export_name: &str,
 		call_data: &[u8],
+	) -> std::result::Result<Vec<u8>, Error> {
+		self.uncached_call_impl(
+			runtime_blob,
+			ext,
+			allow_missing_host_functions,
+			export_name,
+			call_data,
+			&mut None,
+		)
+	}
+
+	/// Same as `uncached_call`, except it also returns allocation statistics.
+	#[doc(hidden)] // We use this function in tests.
+	pub fn uncached_call_with_allocation_stats(
```

Comment:
Why not always return the stats?

Reply:
Are you asking why it's returning an `Option`? Or why isn't there just a single function instead of two where first variant returning the stats and the second not returning them?

(I assume that it's the latter, but I'll explain anyway for other people who might be reading.)

It is an `Option` because in theory there might not be any stats to return if we couldn't call into the runtime at all (e.g. because the passed `export_name` is missing). Otherwise we essentially always want to return those stats: we obviously want to do it when it returns an `Ok`, but we also want to do it when it returns an `Err` (e.g. the runtime panicks due to running out of memory).

So why have two functions? Mostly to reduce churn in this PR. This is *not* the allocation stats PR; I just preemptively added this so that I can write a reasonable test for the issue I'm fixing here. I intend to flesh out the allocation stats (and probably rethink the interface) in a future PR.